src/function.rs

```diff
@@ -63,6 +67,38 @@ extern "C" {
   fn v8__ReturnValue__Get(this: *const ReturnValue) -> *const Value;
 }
 
+// Ad-libbed - V8 does not document ConstructorBehavior.
+/// ConstructorBehavior::Allow creates a regular API function.
+///
+/// ConstructorBehavior::Throw creates a "concise" API function, a function
+/// without a ".prototype" property, that is somewhat faster to create and has
+/// a smaller footprint. Functionally equivalent to ConstructorBehavior::Allow
+/// followed by a call to FunctionTemplate::RemovePrototype().
+#[repr(C)]
+pub enum ConstructorBehavior {
+  Throw,
+  Allow,
+}
+
+/// Options for marking whether callbacks may trigger JS-observable side
+/// effects. Side-effect-free callbacks are allowlisted during debug evaluation
+/// with throwOnSideEffect. It applies when calling a Function,
+/// FunctionTemplate, or an Accessor callback. For Interceptors, please see
+/// PropertyHandlerFlags's kHasNoSideEffect.
+/// Callbacks that only cause side effects to the receiver are allowlisted if
+/// invoked on receiver objects that are created within the same debug-evaluate
+/// call, as these objects are temporary and the side effect does not escape.
+#[repr(C)]
+pub enum SideEffectType {
+  HasSideEffect,
+  HasNoSideEffect,
+  HasSideEffectToReceiver,
+}
+
+#[repr(C)]
+#[derive(Default)]
+pub(crate) struct CFunction([usize; 2]);
```

Comment:
Where does the value 2 come from? I tried adding a size assert for this and it failed.

Reply:
Nevermind - I didn't have the include for v8-fast-api-calls.h